### PR TITLE
Allow promise for locals function in blueprints

### PR DESCRIPTION
**Pull request by knownasilya**
This is a WIP for moving https://github.com/ember-cli/rfcs/pull/7 forward.
Basically you can test it with `ember new [name] --blueprint https://github.com/knownasilya/app-blueprint.git`

The issue is that you cannot select inquirer options using your keys. Basically this is the output:

screen shot 2015-05-06 at 11 31 47 am

"Leaflet" is the default here, and hitting up/down creates those characters below.

### CODE REVIEW

**Comment by knownasilya** [comment]
cc @stefanpenner

**Comment by knownasilya** [comment]
Probably something to do with readline, maybe here: https://github.com/ember-cli/ember-cli/blob/master/lib/ui/index.js#L126

**Comment by knownasilya** [comment]
@MajorBreakfast you contributed to that code, do you have any thoughts about this issue?

**Comment by MajorBreakfast** [comment]
I knew about this bug, but didn't investigate further at the time. My solution was to use one of the other input styles.

**Comment by knownasilya** [comment]
Hum, maybe @SBoudrias has some thoughts.

**Comment by SBoudrias** [comment]
In Inquirer, we mute the input with mute-stream. Maybe something is overridding it?

**Comment by knownasilya** [comment]
@stefanpenner do you have any ideas about this issue?

**Comment by knownasilya** [comment]
@stefanpenner where is the previous ui session defined?

**Comment by knownasilya** [comment]
@stefanpenner where is the "previous UI session"?

**Comment by Turbo87** [comment]
@knownasilya what's the state of this PR? anyone still working on it? is it still relevant?

**Comment by knownasilya** [comment]
I think it's still relevant, and by itself it's pretty much finished. It's the other pieces to the puzzle that need to be figured out.

**Comment by iheanyi** [comment]
This should be fixed by my PR #5621.

**Comment by knownasilya** [comment]
A little stumped why the other tests are failing, the result from `_process` looks fine..

**Comment by stefanpenner** [comment]
#5621 is scheduled for merge

File: lib/models/blueprint.js [hunk 1]
```diff
@@ -845,24 +848,26 @@ Blueprint.prototype._locals = function(options) {
...
-    camelizedModuleName: stringUtils.camelize(sanitizedModuleName),
-    decamelizedModuleName: stringUtils.decamelize(sanitizedModuleName),
-    fileMap: fileMap,
-    hasPathToken: this.hasPathToken,
-    targetFiles: options.targetFiles,
-    rawArgs: options.rawArgs
-  };
-
-  return merge({}, standardLocals, customLocals);
+  return Promise.resolve(this.locals(options))
```
Comment (stefanpenner) [inline lib/models/blueprint.js]: lets use the constructor form here, in-case locals throws.

```js
return new Promise(function(resolve) {
  resolve(this.locals(options);
}.bind(this))
```

**Comment by stefanpenner** [comment]
left 1 comment, once addressed we can merge.

**Comment by knownasilya** [comment]
@stefanpenner change made, but I still don't know why the tests are broken.

**Comment by stefanpenner** [comment]
@homu r+

**Pull request merged**